resolve/resolve.d.ts

```diff
@@ -0,0 +1,131 @@
+// Type definitions for Combokeys v1.1.7
+// Project: https://github.com/substack/node-resolve
+// Definitions by: Ian Clanton-Thuon <https://github.com/iclanton>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace Resolve {
+    interface ResolveFunc {
+        /**
+         * implements the node require.resolve() algorithm such that you can require.resolve() on behalf of a file asynchronously
+         * 
+         * @param {id} a module ID
+         * @param {opts} resolve options
+         * @param {cb} a callback function
+         */
+        (id: string, opts: IResolveOptsAsync, cb: (error: Error, resolution: string) => void): void;
+
+        /**
+         * implements the node require.resolve() algorithm such that you can require.resolve() on behalf of a file asynchronously
+         * 
+         * @param {id} a module ID
+         * @param {cb} a callback function
+         */
+        (id: string, cb: (error: Error, resolution: string) => void): void;
+
+        /**
+         * implements the node require.resolve() algorithm such that you can require.resolve() on behalf of a file synchronously
+         * 
+         * @param {id} a module ID
+         * @param {opts} resolve options
+         * @returns the path to the module's "main" package
+         */
+        sync(id: string, opts?: IResolveOptsSync): string;
+
+        /**
+         * return whether a package is in core
+         * 
+         * @param {id} a module ID
+         * @returns true if the package is in core, otherwise false
+         */
+        isCore(id: string): boolean;
+    }
+
+    interface IResolveOptsAsync extends IResolveOptsBase {
```

Comment:
https://www.typescriptlang.org/docs/handbook/writing-declaration-files.html#naming-conventions
> Naming Conventions
In general, you shouldn’t prefix interfaces with I (e.g. IColor). Because the concept of an interface in TypeScript is much more broad than in C# or Java, the IFoo naming convention is not broadly useful.